File: KitapeviStokTakipWebApi/KitapeviStokTakipWebApi/Models/Book.cs
using Oracle.ManagedDataAccess.Types;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace KitapeviStokTakipWebApi.Models
{
    public class Book
    {
        public int BookId { get; set; }
        public Category Category { get; set; }
        public Publisher Publisher { get; set; }
        public string BookName { get; set; }
        public string Writer { get; set; }
        public int Price { get; set; }
        public int StockAmount { get; set; }
        public string Barcode { get; set; }
        public string AddedDate { get; set; }
    }
}

File: KitapeviStokTakip/src/app/models/book.ts
import { Category } from "./category";
import { Publisher } from "./publisher";

export class Book{
    bookId: number;
    category: Category;
    publisher: Publisher;
    bookName: string;
    writer: string;
    price: number;
    stockAmount: number;
    barcode: string;
    addedDate: string;
}